### ISSUE TITLE
Have useful group order in ACL selector

### ISSUE THREAD

**Issue by LukasKalbertodt** [elan-ev/tobira#1288]
Also something Olaf brought up today: if you have one group per course, then it makes sense to first show groups/courses that are more recent, for example.

But we already discussed that this is low priority, especially because it's kind of hard to implement. Also, users can always search through the groups, so the UX should already be quite good.

**Comment by jjyutm4532** [comment]
Hello, we tried to solve the issue.

This is what we did:

Add a creation_date column to the known_groups table and modify the query to order by this date descending. This will prioritize more recent groups in the results.

You can review changes in this commit: https://github.com/jjyutm4532/elan-ev-tobira-1288/commit/47bf0c93a0f474e402b44fa6d4711b4e909b942e.


> [!CAUTION]
> Disclaimer: The concept of solution was created by AI and you should never copy paste this code before you check the correctness of generated code. Solution might not be complete, you should use this code as an inspiration only.

---

If you no longer want Latta AI to attempt solving issues on your repository, you can block this account.

**Comment by oas777** [comment]
WTF?

**Comment by LukasKalbertodt** [comment]
Yep, exactly my thoughts :D will delete that and report the account, just wanted to have a look at it first. It actually forked this project and suggests a specific code patch (which absolutely does not work btw). 

AI is amazing, bullshit as a service.

**Comment by oas777** [comment]
Reminds me of the ads in old school forums: "You want to fix the brake for you 1983 Cannondale ST500? Try temu.com for a solution!"

**Comment by LukasKalbertodt** [comment]
For reference, to understand our now hidden comments: there was an automated spam comment here before:

![image](https://github.com/user-attachments/assets/f9a1d5c8-4a97-4173-b149-27c6b35a3c97)

**Comment by mpmuc84** [comment]
We came across this issue recently, as we have several (~20) known groups representing university faculties and a few predefined additional groups. Currently, there does not appear to be any sorting going on (I might be wrong about that!), at least we could not see where the order of the list comes from (not the primary role, not any of the labels). The dropdown menu shows the list in the order of the API response, as far as I can see.

Barring a user-controlled option to sort the list: Would it be possible to either set some default sorting for the dropdown or the API response, so that users may have a better way of navigating the list (without having to type to search)?

Granted, a list of 20+ organizational groups is nothing compared to a list of courses (possibly spanning multiple semesters), but the current display of a list completely without any ordering and having to rely on users searching via text input looks not ideal, at least to me.

**Comment by LukasKalbertodt** [comment]
True, these groups are currently not ordered at all, we just load them from the DB and keep whatever order the DB passes them to us. That's certainly not ideal. 

We could just sort them by label, but that's often also not ideal as some groups will be used much more often (e.g. "Everyone", ROLE_ANONYMOUS). In the table showing the already selected groups, we sort by "size", i.e. a group that contains another group is always listed before that other group. This implies ROLE_ANONYMOUS being listed first, followed by ROLE_USER, and then the rest. We can do it like that for the group selector as well. And all groups without "contains" relationship are just sorted by name in their bucket. Would that be a useful solution?

**Comment by mpmuc84** [comment]
This does indeed sound like a helpful solution. I definitely see the advantage of having "larger" or more frequently used groups at the top of the list. Sorting the rest by role name makes sense to me as well, and this way we could achieve some sort of consistency.

Let me know if you need more input from our side.
